03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,334 @@
+plots_dir <- "plots" # Replace with path to desired output plots directory
+
+# Create the plots folder if it doesn't exist
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+## Import and set up data
+Data downloaded from refine.bio include a metadata [tab separated values ("tsv")](https://en.wikipedia.org/wiki/Tab-separated_values) file and a data tsv file.
+This chunk of code will read the both tsv files and add them as data.frame
+objects to your environment.
+
+The data that we downloaded from refine.bio for this analysis has 19 samples
+(obtained from 19 mice with acute myeloid leukemia (AML)), containing
+RNA-sequencing results for types of AML under controlled treatment conditions.
+
+TODO: Add link to the `getting started` section, specifically to point out how
+one can download the dataset.
+
+Now let's import the metadata and count data files.
+
+```{r}
+# Read in metadata tsv file
+metadata <- readr::read_tsv(file.path(
+  "data", # Replace with path to your metadata file
+  "metadata_SRP070849.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data tsv file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP070849.tsv" # Replace with the name of your data file
+)) %>%
+  # Here we are going to store the gene IDs as rownames so that we can have a
+  # numeric matrix to perform calculations on later
+  tibble::column_to_rownames("Gene")
+```
+
+Let's take a look at the metadata object that we read into the R environment.
+
+```{r}
+head(metadata)
+```
+
+Now let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+## How are our samples clustering together?
+
+Now we are going to use a combination of functions from the `DESeq2` and 
+`pheatmap` packages to look at how are samples and genes are clustering.
+
+### Create a DESeqDataset
+We will be using the `DESeq2` package for normalization, which requires us to
+format our data into a `DESeqDataSet` object. We turn the data.frame (or matrix)
+into a [`DESeqDataSet` object](TODO: Add link to header section) and specify
+which variable labels our experimental groups using the `design` argument. 
+
+```{r}
+# The `DESeqDataSetFromMatrix` function needs the values to be converted to
+# integers
+df <- df %>%
+  # Mutate numeric variables to be integers
+  dplyr::mutate_if(is.numeric, round)
+
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = df, # This is the data.frame with the counts values for all replicates in our dataset
+  colData = metadata, # This is the data.frame with the annotation data for the replicates in the counts data.frame
+  design = ~refinebio_treatment # Replace with an appropriate design variable for your analysis
```

Comment:
I think we should probably include the `cancer_type` info in the model too (I haven't read too much about the set up, but it looks like the mutations and the drug treatments are two separate things that are being tested).